packages/truffle-core/lib/commands/migrate.js

```diff
@@ -87,6 +87,10 @@ const command = {
       {
         option: "--skip-dry-run",
         description: "Do not run a test or 'dry run' migration."
+      },
+      {
+        option: "--quiet",
```

Comment:
Maybe if this option is only for compilation we should consider naming it something else? Something like `--quiet-compile`?